src/components/generics/vacation-test-forms/myVacationRequests.tsx

```diff
@@ -1,287 +1,349 @@
-                              onStartWeekChange={() => {
-                                throw new Error("Function not implemented.");
-                              } }
-                              onEndWeekChange={() => {
-                                throw new Error("Function not implemented.");
-                              } }
-                            />
-                          </Box>
-                        </TableCell>
-                        <TableCell>
-                          { renderVacationType() }
-                        </TableCell>
-                        <TableCell>
-                          { renderVacationDaysSpent() }
-                          { renderVacationCommentBox() }
-                          <Box display="flex" justifyContent="center">
-                            { renderVacationApplyButton() }
-                          </Box>
-                        </TableCell>
-                        <TableCell>
-                          <IconButton
-                            onClick={ deleteRequest }
-                            aria-label="delete"
-                            className={ classes.deleteButton }
-                            size="large"
-                          >
-                            <DeleteIcon fontSize="medium"/>
-                          </IconButton>
-                        </TableCell>
-                      </TableRow>
-                    </>
-                  </Collapse>
-                </TableCell>
-              </>
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
+                      { handleRequestStatus(request.hrManagerStatus) }
+                    </StyledTableCell>
+                    <TableCell>
+                      <IconButton
+                        aria-label="expand row"
+                        size="small"
+                        onClick={() => {
+                          const newOpenRows = [...openDetails];
+                          newOpenRows[index] = !newOpenRows[index];
+                          setOpenDetails(newOpenRows);
+                        }}
+                      >
+                        { openDetails[index] ? <KeyboardArrowUpIcon/> : <KeyboardArrowDownIcon/> }
+                      </IconButton>
+                      <IconButton
+                        aria-label="expand row"
+                        size="small"
+                        onClick={() => {
+                          const newOpenRows = [...openEdit];
+                          newOpenRows[index] = !newOpenRows[index];
+                          setOpenEdit(newOpenRows);
+                          setUpdatedVacationRequest({
+                            startDate: request.startDate,
+                            endDate: request.endDate,
+                            type: request.type,
+                            message: request.message,
+                            days: request.days
```

Comment:
This is quite a bit of logic for an onClick, could this be moved into its own function?